### PR TITLE
test(server): extend Lobby testing

### PR DESCRIPTION
**Pull request by xerus2000**
the aim is to consolidate it with the other tests (i.e. RequestTest and the like) and expand coverage

based on #382 because that solves some precursive issues

### CODE REVIEW

**Comment by xerus2000** [comment]
RequestTest has just become a lot more manageable and reliable :tada:

**Comment by xerus2000** [comment]
whoopsie - apparently some tests stopped running, that's why it all looks so smooth -.-

**Comment by xerus2000** [comment]
I've started embracing fancy Kotest matchers with which the tests and failures finally made sense to me ;)
I ran the tests multiple times and am seeing no failures at all anymore

**Comment by xerus2000** [comment]
There are so many things going wrong in the Lobby networking, it's not even funny... ComodificationExceptions are everywhere :/

**Comment by xerus2000** [comment]
This is getting quite big, but I guess I'll go with it - the tests are still not fully fixed

**Comment by xerus2000** [comment]
oh yeah, it's all working!

**Review by SKoschnicke — APPROVED** [review]
So much nicer!

File: server/test/sc/server/network/LobbyRequestTest.kt [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                val reservations = listener.prepareGameResponse.reservations
+                players[1].joinPreparedGame(reservations[0])
+                players[2].joinPreparedGame(reservations[1])
+                await { room.status shouldBe GameRoom.GameStatus.ACTIVE }
+            }
+            
+            val playerListeners = room.slots.map { slot ->
+                PlayerListener().also { listener -> slot.role.player.addPlayerListener(listener) }
+            }
+            "start game on unpause" {
```
[the remark this answers is not in the document]
Comment (xerus2000) [reply server/test/sc/server/network/LobbyRequestTest.kt:75]: it should just behave like any case in which a move is sent without request: the game is canceled with an appropriate error. But yes, maybe that should have a test.

**Pull request merged**